File: app/src/main/java/com/dot2line/vocabar/util/DBUtil.java
package com.dot2line.vocabar.util;

import com.dot2line.vocabar.model.VocaBook;

import org.jetbrains.annotations.NotNull;

import io.realm.Realm;
import io.realm.RealmResults;

public class DBUtil {

  public static final String BOOK_ID = "book_id";

  public static RealmResults<VocaBook> getVocaBookList(Realm realm) {
    return realm.where(VocaBook.class).findAll();
  }

  public static VocaBook getVocaBook(@NotNull Realm realm, @NotNull String bookId) {
    return realm.where(VocaBook.class)
        .equalTo("id", bookId)
        .findFirst();
  }

  public static void removeVocaBook(@NotNull Realm realm, @NotNull String bookId) {
    getVocaBook(realm, bookId).deleteFromRealm();
  }
}

File: app/src/main/java/com/dot2line/vocabar/util/DBUtil.kt
package com.dot2line.vocabar.util

import com.dot2line.vocabar.model.VocaBook

import io.realm.Realm
import io.realm.RealmResults

object DBUtil {

  val BOOK_ID = "book_id"

  fun getVocaBookList(realm: Realm): RealmResults<VocaBook> {
    return realm.where(VocaBook::class.java).findAll()
  }

  fun getVocaBook(realm: Realm, bookId: String): VocaBook? {
    return realm.where(VocaBook::class.java)
        .equalTo("id", bookId)
        .findFirst()
  }

  fun removeVocaBook(realm: Realm, bookId: String) {
    getVocaBook(realm, bookId)?.deleteFromRealm();
  }
}
